drivers/sensor/st/lis2dh/lis2dh.c

```diff
@@ -271,6 +271,21 @@ static int lis2dh_acc_range_set(const struct device *dev, int32_t range)
 }
 #endif
 
+#ifdef CONFIG_LIS2DH_ACCEL_HP_FILTERS
+int lis2dh_acc_hp_filter_set(const struct device *dev, int32_t val)
```

Comment:
Remove the prototype from `lis2dh.h` line 312 and add `static`
```suggestion
static int lis2dh_acc_hp_filter_set(const struct device *dev, int32_t val)
```